.github/workflows/scripts/slack_notifications/slack_notifier.rb

```diff
@@ -0,0 +1,19 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'httparty'
+
+class SlackNotifier
+  def self.send_slack_message(message)
+    path = ENV['SLACK_GEM_NOTIFICATIONS_WEBHOOK']
+    options = {headers: {'Content-Type' => 'application/json'},
+               body: {text: message}.to_json}
+
+    # Pause to avoid slack throttling
+    sleep(1)
```

Comment:
This approach has the great benefit of the sleeping logic residing in a central, shared place here.

It has a two drawbacks that stands out...

1. We delay notifying others about something potentially important by 1 second. This is easily fixed by swapping the `HTTParty.post()` and `sleep()` lines.
2. We sleep even if there is only 1 notification going out for today and even if it has been more than a second since the last message was sent out. This one requires a bit more architecting to fix, and can easily be tabled given the positive benefits previously mentioned.

Reply:
Very nice I didn't think about this! For point #2, I can see how at scale that would be a terrible decision as-is. I've [addressed](https://github.com/newrelic/newrelic-ruby-agent/pull/1432/commits/6b044062a798dca30a0c6649b73684c31246f934#diff-28d6e09aae9f0b8a06eaa86323bb371cd4ea0ab2b399e13d0e89ca760b7bf704R14-R15) point #1, and would love to talk about #2 in the future. I've made note of it.